### PR TITLE
Add GH Actions workflow for running tests

### PR DESCRIPTION
**Pull request by tjurtsch**

### CODE REVIEW

**Comment by tjurtsch** [comment]
It was tested on fork, CI is green: https://github.com/antmicro/ql_fasm/actions

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Run tests
+on:
+  push:
+  pull_request:
+    branches: [ master ]
```
Comment (acomodi) [inline .github/workflows/test.yml]: This is to trigger CI only on pull requests to `master`, right? If so I think we can potentially avoid this restriction and just run GH actions on any PR against any branch.

**Review by acomodi — APPROVED** [review]
Small comment, otherwise LGTM

File: .github/workflows/test.yml (hunk 1 above)
Comment (tjurtsch) [reply .github/workflows/test.yml]: You're right. I removed the limitation

**Comment by mkurc-ant** [comment]
@lnsharma @tpagarani Could you please look and approve this PR ?

**Review by sharmaln — APPROVED** [review]
looks fine

**Comment by mkurc-ant** [comment]
Great, thank you.

**Pull request merged**